adapter/internal/discovery/xds/server.go

```diff
@@ -1239,3 +1300,13 @@ func UpdateEnforcerThrottleData(throttleData *throttle.ThrottleData) {
 	enforcerThrottleData = t
 	logger.LoggerXds.Infof("New Throttle Data cache update for the label: " + label + " version: " + fmt.Sprint(version))
 }
+
+func updateLCStateOfMgwSwagger(mgwSwagger *model.MgwSwagger) {
+	// If there are any metadata stored under the APIMetadataMap and Life Cycle state is blocked, update the mgwSwagger
+	apiEntry, apiFound := APIMetadataMap[mgwSwagger.GetID()]
+	if apiFound && apiEntry.LcState == blockedStatus {
+		mgwSwagger.LifeCycleState = blockedStatus
+		return
+	}
+	mgwSwagger.LifeCycleState = nonBlockedStatus
```

Comment:
Leave the lifecycleState as it is.

Reply:
Changed the lifecycle status to be published.

Reply:
Reverted to previous state as we cannot decide whether the API is in created or published state when deployed for the first time. (Cannot depend on the API.yaml to populate this field)